src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.Windows.cs

```diff
@@ -174,24 +175,28 @@ private async void OnWebResourceRequested(CoreWebView2 sender, CoreWebView2WebRe
 			}
 
 			// 3.b. Otherwise, return a 404
-			var ras404 = new InMemoryRandomAccessStream();
-			using (var writer = new StreamWriter(ras404.AsStreamForWrite()))
-			{
-				writer.WriteLine("Resource not found (404)");
-			}
-
+			var ras404 = await CopyContentToRandomAccessStreamAsync(_404MessageBuffer.Value);
 			return (Stream: ras404, ContentType: "text/plain", StatusCode: 404, Reason: "Not Found");
+		}
 
-			static async Task<IRandomAccessStream> CopyContentToRandomAccessStreamAsync(Stream content)
-			{
-				using var memStream = new MemoryStream();
-				await content.CopyToAsync(memStream);
-				var randomAccessStream = new InMemoryRandomAccessStream();
-				await randomAccessStream.WriteAsync(memStream.GetWindowsRuntimeBuffer());
-				return randomAccessStream;
-			}
+		static async Task<IRandomAccessStream> CopyContentToRandomAccessStreamAsync(Stream content)
+		{
+			var ras = new InMemoryRandomAccessStream();
```

Comment:
Both methods have similar logic for creating an InMemoryRandomAccessStream, could we refactor it creating a helper method?
```
private static async Task<IRandomAccessStream> CreateRandomAccessStreamAsync(Func<IRandomAccessStream, Task> writeContentAsync)
{
	var ras = new InMemoryRandomAccessStream();
	await writeContentAsync(ras);
	await ras.FlushAsync();
	ras.Seek(0);
	return ras;
}
```